Applications/ApplicationsLib/LinearSolverLibrarySetup.h

```diff
@@ -0,0 +1,64 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef APPLICATIONSLIB_LINEARSOLVERLIBRARYSETUP_H_
+#define APPLICATIONSLIB_LINEARSOLVERLIBRARYSETUP_H_
+
+namespace ApplicationsLib
+{
+
+/// The LinearSolverLibrarySetup takes care of proper initialization and
+/// shutting down of an external linear solver library. The concrete
+/// implementation is chosen by the build system.
+/// An object of this class must be created at the begining of the scope where
+/// it is used. When the scope closes (or the object is destroyed explicitly)
+/// library shutting down functions are automatically called.
+/// The default implementation is empty providing polymorphic behaviour when
+/// using this class.
+
+#if defined(USE_PETSC)
+#include <petsc.h>
+#include <mpi.h>
+struct LinearSolverLibrarySetup final
```

Comment:
PETSc is not only for linear solver, and also MPI can be also initialized here as an individual option. Therefore, the term of  LinearSolverLibrarySetup is not adequate.